tools/build_defs/repo/jvm.bzl

```diff
@@ -59,6 +65,16 @@ _PASS_PROPS = (
 
 _FETCH_SOURCES_ENV_VAR = "BAZEL_JVM_FETCH_SOURCES"
 
+def _get_auth(ctx, urls): # duplicated from http.bzl
```

Comment:
I think that moving this to `utils.bzl` would be useful, not only for this PR but also for other rules that want to reuse the same logic.

Reply:
Agreed. Is it ok to do this as part of this PR?

Reply:
I would personally say yes, but will defer to a team member. Cc @meteorcloudy

Reply:
@fmeum noticed something:
```
    if srcurls and _should_fetch_sources_in_current_env(repository_ctx):
        repository_ctx.download(
            srcurls,
            srcpath,
            srcsha,
            canonical_id = repository_ctx.attr.canonical_id,
            auth = get_auth(repository_ctx, srcurls),
        )
```

Should this be:
```
canonical_id = repository_ctx.attr.canonical_id or get_default_canonical_id(repository_ctx, srcurls),
```

Reply:
Probably yes, I think I just missed this rule when I made the canonical ID change.